src/Concurrent/BoundedQueue.purs

```diff
@@ -38,6 +40,14 @@ new size = do
   writePos ← AVar.new 0
   pure (BoundedQueue { size, contents, readPos, writePos })
 
+-- | Synchronously creates a new `BoundedQueue` with the given capacity.
+new' ∷ ∀ a. Int → Effect (BoundedQueue a)
```

Comment:
if you remove current `new`
and change this one to
```
new ∷ ∀ a. MonadEffect m => Int → m (BoundedQueue a)
new size = liftEffect do
```

you would get best of both worlds, as it would not brake existing code and will work with Effect

Reply:
Introducing a type parameter can break existing code because then that parameter has to be inferred and inference is not always possible.